### PR TITLE
Gelu reduced precision

### PR DESCRIPTION
**Pull request by mikowals**
`gelu(x)` on X10 fails when x is reduced precision.  I added a check on x and reduce precision of all the tensor constants it uses if necessary.

I also implemented tests (based off leakyRelu) in X10/ops_tests.swift.  **I have not run the tests I added yet** as I don't know how to trigger them locally (Xcode or command line in MacOS).  I think this is the correct place for them and they were previously omitted maybe because `gelu()` is not reimplemented in X10's rawOpsManual.swift.

### CODE REVIEW

**Comment by asuhan** [comment]
Cool! I'll defer to @dan-zheng on matters of style, the logic looks correct to me.

**Comment by mikowals** [comment]
I found the `Tensor(... , deviceAndPrecisonLike: )` so I used it to clean up the code.

I added a gradient test for `gelu()` because I initially didn't realise `withoutDerivative` was needed and all tests passed still passed.  But backprop failed in a network locally.  So I added a test that would have caught it.

File: Sources/TensorFlow/Operators/Math.swift [hunk 1]
```diff
@@ -1365,14 +1365,16 @@ func _vjpElu<T: TensorFlowFloatingPoint>(
 @inlinable
 @differentiable
 public func gelu<T: TensorFlowFloatingPoint>(_ x: Tensor<T>) -> Tensor<T> {
-  let ratio1 = Tensor<T>(0.7978845608, on: x.device)  // An approximation of √(2/π).
+  let xWithoutDerivative = withoutDerivative(at: x) { $0 }
```
Comment (dan-zheng) [inline Sources/TensorFlow/Operators/Math.swift:1368]: Is it necessary to use `withoutDerivative(at: x)` instead of `x` directly here?

Comment (dan-zheng) [inline Sources/TensorFlow/Operators/Math.swift:1368]: If `withoutDerivative(at: x)` is necessary, you can simplify it to this version:
```suggestion
  let xWithoutDerivative = withoutDerivative(at: x)
```

Comment (mikowals) [reply Sources/TensorFlow/Operators/Math.swift:1368]: Thanks. I did not realise I could write it that way.  I will be able to use `x` directly once #963 is merged.

Comment (mikowals) [reply Sources/TensorFlow/Operators/Math.swift]: done now.

**Comment by mikowals** [comment]
I have messed up git.  I will close this and make another pull request.
